src/components/KFileUpload/KFileUpload.vue

```diff
@@ -0,0 +1,384 @@
+    },
+    label: {
+      type: String,
+      default: '',
+    },
+    /**
+     * Test mode - for testing only, strips out generated ids
+     */
+    testMode: {
+      type: Boolean,
+      default: false,
+    },
+    help: {
+      type: String,
+      default: undefined,
+    },
+    buttonAppearance: {
+      type: String,
+      default: 'primary',
+    },
+    buttonText: {
+      type: String,
+      default: 'Select file',
+    },
+    fileModel: {
+      type: String,
+      default: undefined,
+    },
+    removable: {
+      type: Boolean,
+      default: true,
+    },
+    placeholder: {
+      type: String,
+      default: 'No file selected',
+    },
+    /**
+     * Set whether its file upload or image upload type
+     */
+    type: {
+      type: String,
+      default: 'file',
+      validator: (value: string): boolean => {
+        return ['file', 'image'].includes(value)
+      },
+    },
+    accept: {
+      type: Array as PropType<string[]>,
+      required: true,
+    },
+    maxFileSize: {
+      type: Number,
+      default: null,
+    },
+    /**
+     * Set icon size
+     */
+    iconSize: {
+      type: String,
+      default: '26',
+    },
+    icon: {
+      type: String,
+      default: 'image',
+    },
+    /**
+     * Set icon color
+     */
+    iconColor: {
+      type: String,
+      default: undefined,
+    },
+    hasError: {
+      type: Boolean,
+      default: false,
+    },
+    errorMessage: {
+      type: String,
+      default: 'Please check file size.',
+    },
+  },
+
+  emits: ['file-added', 'file-removed', 'error'],
+
+  setup(props, { emit }) {
+    const customInputId = computed((): string => props.testMode ? 'test-file-upload-id-1234' : uuidv1())
+    const maximumFileSize = computed((): Number => {
+      if (props.maxFileSize || props.maxFileSize === 0) {
+        return props.maxFileSize
+      }
+      return props.type === 'file' ? 5250000 : 1000000
+    })
+
+    const hasUploadError = ref(false)
+
+    // This holds the FileList
+    const fileInput = ref<File[]>([])
+    // To clear the input value after reset
+    const fileInputKey = ref(0)
+    // File fakepath
+    const fileValue = ref('')
+    // Array to store the previously selected FileList when user clicks reopen the file uploader and clicks on Cancel
+    const fileClone = ref<File[]>([])
+
+    const onFileChange = (evt: any): void => {
+      fileInput.value = evt.target?.files
+      fileValue.value = fileInput?.value[0]?.name
+
+      const fileSize = fileInput?.value[0]?.size
+
+      if (props.type === 'file') {
+        hasUploadError.value = fileSize > maximumFileSize.value
+
+        if (hasUploadError.value) {
+          fileInputKey.value++
+          emit('error', fileInput.value)
+        }
+      } else {
+        hasUploadError.value = fileSize > maximumFileSize.value
+
+        if (hasUploadError.value) {
+          fileInputKey.value++
+          emit('error', fileInput.value)
+        }
+      }
```

Comment:
It looks like the logic inside this `if` block is doing the same thing regardless of the type.

Can we remove the `if` and just use 1 set of logic?